src/braintree/initBraintreePayPalButtons.ts

```diff
@@ -0,0 +1,310 @@
+        },
+    });
+    batchRequests.push({
+        apiType: 'setBillingAddress',
+        payload: {
+            first_name: result.details.firstName,
+            last_name: result.details.lastName,
+            address_line_1: billingAddress.line1,
+            address_line_2: billingAddress.line2 ?? '',
+            business_name: '',
+            city: billingAddress.city,
+            country: billingCountry.name,
+            country_code: billingCountry.iso_code,
+            province: billingProvince.name,
+            province_code: billingProvince.iso_code,
+            phone_number: result.details.phone,
+            postal_code: billingAddress.postalCode,
+        },
+    });
+    shippingOption && batchRequests.push({
+        apiType: 'changeShippingLines',
+        payload: {
+            index: shippingOption.id,
+        },
+    });
+    batchRequests.push({
+        apiType: 'setTaxes',
+        payload: {},
+    });
+    
+    let resp = await batchRequest(batchRequests);
+    /* istanbul ignore next */
+    if (!resp.success) {
+        displayError('There was an unknown error while processing your payment.', 'payment_gateway', 'unknown_error');
+        return;
+    }
+
+    resp = await addPayment({
+        gateway_public_id: payment.properties.public_id,
+        token: result.nonce,
+    });
+    /* istanbul ignore next */
+    if (!resp.success) {
+        displayError('There was an unknown error while processing your payment.', 'payment_gateway', 'unknown_error');
+        return;
+    }
+
+    orderProcessing();
+};
+
+const handleShippingChange = (
+    paypalCheckout: IBraintreePaypalCheckoutInstance
+) => async (
+    data: OnShippingChangeData,
+    actions: OnShippingChangeActions
+): Promise<void> => {
+    /* istanbul ignore next */
+    if (!data.shipping_address) {
+        return actions.reject();
+    }
+    
+    const currency = getCurrency().iso_code;
+    const resp = await estimateShippingLines({
+        city: data.shipping_address.city,
+        country_code: data.shipping_address.country_code,
+        province_code: data.shipping_address.state,
+        postal_code: data.shipping_address.postal_code,
+    } as unknown as IAddress);
+    /* istanbul ignore next */
+    if (!resp.success) {
+        actions.reject();
+        return;
+    }
+
+    let hasSelected = false;
+    const amountDue = getTotals().totalAmountDue;
+    const respData = (resp.response as IApiSuccessResponse).data as IEstimateShippingLinesResponse;
+    const serverSelectedShippingId = respData.application_state.shipping.selected_shipping?.id;
+    const options = respData.application_state.shipping.available_shipping_lines.map(o => {
+        hasSelected ||= serverSelectedShippingId === o.id;
+        return {
+            id: o.id, 
+            label: o.description,
+            selected: serverSelectedShippingId === o.id,
+            amount: {
+                value: (o.amount / 100).toFixed(2),
```

Comment:
I know right now we're only serving the US, but if we have the has_decimal property on currency (or some such property), it would be good to check that to see whether division is required.

Reply:
Fixed